src/Microsoft.TemplateEngine.Cli/OptionalWorkloadPackages.cs

```diff
@@ -0,0 +1,72 @@
+using Microsoft.DotNet.TemplateLocator;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.PhysicalFileSystem;
+using Microsoft.TemplateEngine.Abstractions.TemplatePackage;
+using Microsoft.TemplateEngine.Cli;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Mount.Archive;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.TemplateEngine.Utils
+{
+    internal class OptionalWorkloadPackages : ITemplatePackageProviderFactory
```

Comment:
The name of this class does not look fine to me. The content does not seem like it's any "packages", it is a provider factory. Maybe `OptionalWorkloadPackagesProviderFactory`?

Reply:
Done.